src/test/python/django/transaction_test.py

```diff
@@ -80,6 +80,20 @@ def test_nested_atomic():
     singer2.save()
   print('Atomic Successful')
 
+def test_error_during_transaction():
+  try:
+    transaction.set_autocommit(False)
+
+    singer = Singer(singerid=1, firstname='hello', lastname='world')
+    singer2 = Singer(singerid=2, firstname='hello', lastname='python')
+
+    singer.save()
+
+    raise Exception("Some error has occurred")
```

Comment:
We should not raise an exception in the application itself, but simulate an error in Cloud Spanner. The reason for this is that PostgreSQL has a very strict model when it comes to handling database errors in transactions:
1. A database error in a transaction will invalidate that transaction, and all further statements in that transaction will also fail with the error `current transaction is aborted, commands ignored until end of transaction block`.
2. The only way to get out of that state is to roll back the transaction (or roll back to a savepoint in the transaction, but that is not supported in PGAdapter at the moment).

We should therefore verify that PGAdapter in combination with DJango really shows that behavior.

See this example:
1. `gorm` code: https://github.com/GoogleCloudPlatform/pgadapter/blob/8a19695373ed5371e9baea4a5980b10c50736403/src/test/golang/pgadapter_gorm_tests/gorm.go#L527
2. Corresponding Java test case: https://github.com/GoogleCloudPlatform/pgadapter/blob/8a19695373ed5371e9baea4a5980b10c50736403/src/test/java/com/google/cloud/spanner/pgadapter/golang/GormMockServerTest.java#L780

Reply:
Okay got it. Changed the code.